Assets/HoloToolkit/Input/README.md

```diff
@@ -3,50 +3,649 @@
+- `IHoldHandler` for the Windows hold gesture.
+- `IManipulationHandler` for the Windows manipulation gesture.
+- `INavigationHandler` for the Windows navigation gesture.
 
-The **input manager** listens to the various events coming from the input sources, and also takes into account the gaze. Currently, that gaze is always coming from the GazeManager class, but this could be extended to support multiple gaze sources if the need arises.
+### Motion Controller Specific Interfaces
 
-By default, input events are sent to the currently focused game object, if that object implements the appropriate interface. Modals input handlers can also be added to the input manager: these modal handlers will take priority over the currently focused object Fallback handlers can also be defined, so that the application can react to global inputs that aren't targeting a specific element. Any event sent by the input manager always bubbles up from the object to its ancestors. 
+- `IControllerInputHandler` for input position events from the thumbstick and TouchPad.
+- `ISelectHandler` for selection pressed amount changes.
+- `IControllerTouchpadHandler` for TouchPad press and touch events.
+- `ISourcePositionHandler` to get pointer and grip position events.
+- `ISourceRotationHandler` to get pointer and grip rotation events.
 
-In recap, the **input manager** forwards the various input sources events to the appropriate game object, using the following order:
+## Getting Started
 
-1. The registered modal input handlers, in LIFO (Last-In First-Out) order of registration
-2. The currently focused object
-3. The fallback input handlers, in LIFO order of registration
+First, it’s important to read and review Microsoft’s developer guidelines for [Gaze](https://developer.microsoft.com/en-us/windows/mixed-reality/gaze), [Gestures](https://developer.microsoft.com/en-us/windows/mixed-reality/gestures), [Voice](https://developer.microsoft.com/en-us/windows/mixed-reality/voice_input), and [Motion Controllers](https://developer.microsoft.com/en-us/windows/mixed-reality/motion_controllers).  When setting up input for your own projects you’ll need to keep a few things in mind.
+
+- The InputManager.prefab is required
+- A cursor that inherits from the ICursor interface is required.
+- The `GameObect`(s) you want to interact with are required to have a Collider component.
+- When using uGUI, all world space canvases are required to use the UIRaycastCamera as their event camera.
+- When consuming events in your input handlers, it’s highly suggested you call `eventData.Use();` so the event doesn’t fall through to other input handlers.
+
+### Detecting Input Sources
+
+1. Create a new scene
+2. Run the MRTK scene wizard via `MixedRealityToolkit/Configure/Apply Scene Settings`
+3. Create an empty GameObject
+4. Rename the GameObject to InputSourceHandler
+5. Attach a `SetGlobalListener` component to the InputSourceHander GameObject
```

Comment:
Related to my comment above: I don't think we should encourage people to use the global listener mechanism. They really should always use "regular" listener, or fallback if they want to globally get all unhandled events when not targeted

Reply:
Oh this is a very good point. I'll update this appropriately.  We should have a section exclusively on how to get events and how they're routed.